packages/SwingSet/src/kernel/vat-warehouse.js

```diff
@@ -63,6 +63,8 @@ export function makeVatWarehouse(kernelKeeper, vatLoader, policyOptions) {
     // Often a large contract evaluation is among the first few deliveries,
     // so let's do a snapshot after just a few deliveries.
     snapshotInitial = 2,
+  } = policyOptions || {};
+  let {
```

Comment:
This is making me think more carefully about how `warehousePolicy` is provided. Currently, `makeSwingsetController()` can provide a different `warehousePolicy` each time, as part of the third `runtimeOptions` argument. The "runtime" in the name implies that it's safe to change the value each time you launch the kernel.

That's true for `maxVatsOnline` (which might cause vats to be paged out at a different schedule, but won't change the timing of consensus-visible `writeSnapshot` operations). But both `snapshotInitial` and `snapshotInterval` *do* change that timing. (They're consensus-visible because writing a heap snapshot starts with a `forceGC`, and that changes the heap, which changes the snapshots, and we're still hoping to get consistency of snapshot contents).

So, really, it was never ok to include `snapshotInitial` or `snapshotInterval` in `runtimeOptions`. Instead, we should have accepted initial values in `config`, as passed to `initializeSwingset`, and stored them in the kernelKeeper. Then `makeVatWarehouse` should read them from `kernelKeeper` instead of taking them as arguments.

This overlaps with a likely bug in this code: if there's nothing to persist `snapshotInterval` in the DB, then a sequence of:
* launch kernel with `runtimeOptions` that omits `snapshotInterval`
  * `makeVatWarehouse()` then uses the default of 200
  * host app does `c.changeKernelOptions({snapshotInterval: 50})`
* a few blocks go by
* kernel is rebooted
  * host app is still not overriding `runtimeOptions`
  * `makeVatWarehouse` reverts to default 200

That makes the snapshot-creating schedule dependent upon when exactly the application gets rebooted, and that's a recipe for consensus failure.

So I think we should probably clean this up as part of this PR. Your addition of `setSnapshotInterval` is fine (although it needs to write the value into `kernelKeeper` too, to fix the problem above). But `makeVatWarehouse` should do two reads to get the values to hold in RAM, and `initializeSwingset.js`/`initializeKernel.js` should remove them from the acceptable `runtimeOptions` values, and `initializeKernel()` should copy them from `config` into `kernelKeeper` somewhere. These default numbers will live in `initializeKernel()`.

Reply:
Yeah, as I was updating `configuration.md` I noticed that we had decided to make `heapSnapshotInterval` updatable but it isn't one of the kernel configuration properties you can set initially, and that definitely seemed like something was off.  I noticed that the vat warehouse keeps the interval value in RAM but I unthinkingly just thought it was fetched by some other code somewhere higher on the call stack, but I didn't immediately sniff that that was self evidently a wrong thought.

I agree with your analysis and will make the appropriate updates as part of this PR.